internal/backends/collection.go

```diff
@@ -315,12 +318,15 @@ func (cc *collectionContract) ListIndexes(ctx context.Context, params *ListIndex
 	res, err := cc.c.ListIndexes(ctx, params)
 	checkError(err, ErrorCodeCollectionDoesNotExist)
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/3589
-	// if res != nil && len(res.Indexes) > 0 {
-	// 	must.BeTrue(slices.IsSortedFunc(res.Indexes, func(a, b IndexInfo) int {
```

Comment:
That code was correct – we should only verify that results are sorted, not actually sort them. That's because `ListIndexes` implementation could be called by other methods of that implementation, and we want them to be able to rely on the fact that the list is sorted.

Reply:
but the code was panicking because `.IsSortedFunc` return false

Reply:
Yes, that's why we need to update implementations of this method:
* https://github.com/FerretDB/FerretDB/blob/d7493f75f0d8d9109dd3cc816e529425fea93cd6/internal/backends/postgresql/collection.go#L441
* https://github.com/FerretDB/FerretDB/blob/d7493f75f0d8d9109dd3cc816e529425fea93cd6/internal/backends/sqlite/collection.go#L415

Reply:
I see, I understood now, should on the pg and sqlite handler

Reply:
do you still need this after the sort process?

https://github.com/FerretDB/FerretDB/blob/d7493f75f0d8d9109dd3cc816e529425fea93cd6/internal/backends/postgresql/collection.go#L486

Reply:
We need to uncomment those lines and fix tests

Reply:
could you check this: https://github.com/FerretDB/FerretDB/commit/c7a7f059d10eab28805db8dd27d44217b4cbcde3

I will also remove the test

Reply:
That commit looks correct, but we need to fix integration tests after that